plugin.js

```diff
@@ -13,19 +14,22 @@ export default (ctx) => {
 
   const { isDev, isClient, isServer, app, route, beforeNuxtRender, store } = ctx
 
-  <% Object.keys(options.networkInterfaces).forEach((key) => { %>
-    let networkInterface = require('<%= options.networkInterfaces[key] %>')
-    networkInterface = networkInterface.default(ctx) || networkInterface(ctx)
+  <% Object.keys(options.clientConfigs).forEach((key) => { %>
+    let client = require('<%= options.clientConfigs[key] %>')
+    // es6 module default export or not
+    client = client.default(ctx) || client(ctx)
 
     const opts = isServer ? {
-        ssrMode: true
+        ssrMode: true,
+        cache: new InMemoryCache()
```

Comment:
Why creating a new cache and not getting the one from the config?

Reply:
fasho

Reply:
You are still unnecessary creating a cache when one is already provided.

Reply:
The code should look like this:

```js
const cache = client.cache || new InMemoryCache()

const opts = isServer ? {
  ssrMode: true,
} : {
  ssrForceFetchDelay: 100,
  connectToDevTools: isDev
}

if (!isServer) {
  cache.restore(window.__NUXT__ ? window.__NUXT__.apollo.<%= key === 'default' ? 'defaultClient' : key %> : null)
}

const finalOptions = Object.assign({}, opts, client, { cache })
```

Reply:
ok, yea, using the same singleton, I get what you're saying.